build/local_builder.go

```diff
@@ -135,21 +145,73 @@ func (l *LocalBuilder) PullBaseImageNotExist() (err error) {
 	if l.Image.Spec.Layers[0].Value == common.ImageScratch {
 		return nil
 	}
-	if err = image.NewImageService().PullIfNotExist(l.Image.Spec.Layers[0].Value); err != nil {
+	if err = l.ImageService.PullIfNotExist(l.Image.Spec.Layers[0].Value); err != nil {
 		return fmt.Errorf("failed to pull baseImage: %v", err)
 	}
 	logger.Info("pull baseImage %s success", l.Image.Spec.Layers[0].Value)
 	return nil
 }
 
+func (l *LocalBuilder) IsHashConsistent(layer *v1.Layer, cacheid string) bool {
+	if cacheid == "" {
+		return false
+	}
+
+	baseDir, err := os.Getwd()
+	if err != nil {
+		logger.Error("failed to get current path, err: %v", err)
+		return false
+	}
+
+	layerHash, err := hash.CheckSum(filepath.Join(baseDir, strings.Fields(layer.Value)[0]))
```

Comment:
why checksum under currentDir?

Reply:
I guess it is a build context, how about put it within builder, get it before building?